src/tests/IntegrationTest.t.sol

```diff
@@ -82,4 +108,99 @@ abstract contract IntegrationTest is DSSTest {
         assertEq(address(ilk.clip), address(0));
     }
 
+    function test_optimism_relay() public {
+        DaiAbstract l2Dai = DaiAbstract(0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1);
+        OptimismDaiBridgeLike l2Bridge = OptimismDaiBridgeLike(0x467194771dAe2967Aef3ECbEDD3Bf9a310C76C65);
+        mcd.dai().setBalance(address(this), 100 ether);
+        OptimismDaiBridgeLike bridge = OptimismDaiBridgeLike(mcd.chainlog().getAddress("OPTIMISM_DAI_BRIDGE"));
+
+        // Transfer some DAI across the Optimism bridge
+        mcd.dai().approve(address(bridge), 100 ether);
+        bridge.depositERC20To(address(mcd.dai()), address(l2Dai), address(this), 100 ether, 1_000_000, "");
+
+        // Message will be queued on L1, but not yet relayed
+        assertEq(mcd.dai().balanceOf(address(this)), 0);
+
+        // Relay the message
+        optimism.relayFromHost();
```

Comment:
Wondering if implicitly stay in the rollup might be confusing and if maybe we should switch back to the sender chain within this function (same for the other way). Then in the test you need to explicitly force back the switch if you want to test something in the receipting of the message.
Might be unnecessary though as you want to probably always test something on the recipient chain but not sure. Maybe some renaming can clarify the behaviour.

Reply:
Yeah I originally had it like this, but I found it would almost always just end up bring relay followed by a switch. What about adding a `bool selectFork` as an argument where we would almost always just pass `true`?

Reply:
That seems a good idea.

Reply:
added